generate-event.go

```diff
@@ -1,367 +1,158 @@
 package main
 
 import (
-	"encoding/base64"
-	"fmt"
-	"os"
-	"text/template"
-
+	"log"
 	"github.com/codegangsta/cli"
+	"strings"
+	"fmt"
+	"encoding/json"
+	"regexp"
 )
 
-var events = map[string]string{
-	"S3":       s3Event,
-	"SNS":      snsEvent,
-	"Kinesis":  kinesisEvent,
-	"DynamoDB": dynamodbEvent,
-	"Api":      apiEvent,
-	"Schedule": scheduleEvent,
+/**
+    The go:generate tag is used to generate a binary go file that
+    contains all data from the sample-events folder, which is used
+    by this file to generate commands for generate-event. If any changes
+    are made to the sample-events folder, the binary file will need to be
+    re-generated via running
+        'go generate'
+    in the project root folder. This command requires go-bindata, which can be
+    installed by running
+        'go get -u github.com/jteeuwen/go-bindata/...'
+ */
+// ***DO NOT EDIT THE FOLLOWING COMMENT - it is required for 'go generate' to work***
+//go:generate go-bindata -o generated-event-binary.go sample-events/...
+
+var nameToCommandMap map[string]string = make(map[string]string)
+var nameToDescriptionMap map[string]string = make(map[string]string)
+var nameToFlagListMap map[string][]Flag = make(map[string][]Flag)
+
+type CommandMap struct {
+	Commands []Command
 }
 
-func generate(eventType string, c *cli.Context) {
+type Command struct {
+	Name        string `json:"name"`
+	Command     string `json:"command"`
+	Description string `json:"description"`
+	Flags       []Flag `json:"flags"`
+}
 
-	sample, ok := events[eventType]
-	if !ok {
-		fmt.Printf("Unsupported event type: %s", eventType)
-		os.Exit(1)
-	}
+type Flag struct {
+	Name    string `json:"name"`
+	Usage   string `json:"usage"`
+	Value   string `json:"value"`
+}
 
-	t, err := template.New("event").Parse(sample)
+// Generates top-level commands for generate-event for all
+// categories under sample-events
+func generateEventCategories() (categories []cli.Command) {
+	mapFileToCommands()
+
+	dirs, err := AssetDir("sample-events")
 	if err != nil {
-		fmt.Printf("Failed to load sample %s event: %s", eventType, err)
-		os.Exit(1)
+		log.Fatal(err)
+	}
+
+	for _, dir := range dirs {
+		if (strings.Contains(dir, ".")) {
```

Comment:
hmm.. you should use Go's neat APIs to do this check for you - https://stackoverflow.com/questions/8824571/golang-determining-whether-file-points-to-file-or-directory